src/auth/authApp.js

```diff
@@ -84,23 +82,123 @@ export function redirectToSignInWithAuthRequest(authRequest: string = makeAuthRe
     return
   }
 
-  function successCallback() {
-    Logger.info('protocol handler detected')
-    // protocolCheck should open the link for us
+  // Create a unique ID used for this protocol detection attempt.
+  const echoReplyID = Math.random().toString(36).substr(2, 9)
```

Comment:
Is this supposed to be cryptographically secure?  What happens if someone (e.g. an attacker) is able to guess this value?  Can they hijack the redirect?

Reply:
No, it's supposed to create a unique "auth" session in the situation that the user clicks "sign in with blockstack" on the same web app in several different tabs - or some other situation similar to that. This is the unique ID stored in localStorage for the duration of the protocol detection "ping".